hooks/hooks.js

```diff
@@ -0,0 +1,102 @@
+import {useFetch} from "./useFetch"
+
+// Gateway API
+// Manage the Corona Network P2P gateway
+export const useNetworkStatus = () =>{
+  return useFetch('https://corona-network/gateway', "GET", null)
+}
+export const useConnect = () =>{
+  return useFetch('https://corona-network/gateway', "PUT", null)
+}
+export const useDisconnect = () =>{
+  return useFetch('https://corona-network/gateway', "DELETE", null)
+}
+
+
+// Profile API 
+// Manage the local Corona Network profile
+export const useNewProfile = () =>{
+  return useFetch('https://corona-network/profile', "POST", null)
+}
+export const useGetProfile = () =>{
+  return useFetch('https://corona-network/profile', "GET", null)
+}
+export const useUpdateProfile = (body={"name": "test"}) =>{
+  return useFetch('https://corona-network/profile', "PUT", body)
+}
+export const useDeleteProfile = () =>{
+  return useFetch('https://corona-network/profile', "DELETE", null)
+}
+// Manage the local Corona Network Avatar
+export const useGetAvatar = () =>{
+  return useFetch('https://corona-network/profile/avatar', "GET", null)
+}
```

Comment:
Retrieving an avatar is not really an API call. Rather it is a direct image retrieval. This call will do a temporary redirect to a `cdn` url which will serve the actual image. Long story short, it's not a JSON response thing, rather an img source.